disk/test/com/treode/disk/edit/SimpleLogSpec.scala

```diff
@@ -0,0 +1,98 @@
+package com.treode.disk
+
+import java.nio.file.{Paths, StandardOpenOption}
+import com.treode.async.{Async, Scheduler}
+import com.treode.async.io.File
+import com.treode.async.io.stubs.StubFile
+import com.treode.async.stubs.StubScheduler
+import scala.util.{Failure, Success}
+import com.treode.buffer.PagedBuffer
+import com.treode.async.Async
+import com.treode.async.Async.async
+import org.scalatest._
+import com.treode.async.implicits._
+import com.treode.async.stubs.implicits._
+import com.treode.async.Globals
+
+class SimpleLog (file: StubFile, geom: DriveGeometry) (implicit scheduler: Scheduler) {
+   import geom.{blockAlignUp, blockAlignDown}
+   
+   var pos = 0L
+   val buf = PagedBuffer (10)
+   
+   def record (s: String): Async [Unit] =
+   async { cb =>
+      // read pos is aligned at start, and after each record
+      // write pos is at end of last record
+      if (buf.writePos > 0) {
+         buf.writePos -= 1
+         buf.writeByte(1)
+      }
+      buf.writeString (s)
+      buf.writeByte(0)
```

Comment:
This way, `end` is one byte after the end of the last record. If you move buf.writeByte(0) in between line 34 and 35, then `end` will be the end of the last record. Then you won't need to conditionally subtract 1 from writePos above.